x-pack/plugins/observability_solution/observability_onboarding/public/application/onboarding_flow_form/onboarding_flow_form.tsx

```diff
@@ -83,18 +83,20 @@ export const OnboardingFlowForm: FunctionComponent = () => {
   const radioGroupId = useGeneratedHtmlId({ prefix: 'onboardingCategory' });
 
   const [searchParams, setSearchParams] = useSearchParams();
-  const packageListSearchBarRef = React.useRef<null | HTMLInputElement>(null);
+  const packageListRef = React.useRef<HTMLDivElement | null>(null);
   const [integrationSearch, setIntegrationSearch] = useState('');
 
   const createCollectionCardHandler = useCallback(
     (query: string) => () => {
       setIntegrationSearch(query);
-      if (packageListSearchBarRef.current) {
-        packageListSearchBarRef.current.focus();
-        packageListSearchBarRef.current.scrollIntoView({
-          behavior: 'auto',
-          block: 'center',
-        });
+      if (packageListRef.current) {
+        // adding a slight delay causes the search bar to be rendered
+        new Promise((r) => setTimeout(r, 10)).then(() =>
+          packageListRef.current?.scrollIntoView({
+            behavior: 'smooth',
+            block: 'center',
+          })
+        );
       }
     },
     [setIntegrationSearch]
```

Comment:
I don't have a good idea to go around the timeout, I think that's the simples solution here. But we could wrap this into a custom hook to make the code a bit nicer, something like:
```tsx
function usePackageListScrollTrigger(packageList: HTMLDivElement | null) {
  const [scrollTrigger, setScrollTrigger] = useState<number | null>(null);

  useEffect(() => {
    if (packageListRef.current === null || scrollTrigger === null) {
      return;
    }

    const timeout = setTimeout(() => {
      packageListRef.current?.scrollIntoView({
        behavior: 'smooth',
        block: 'center',
      });
    }, 100);

   return () => clearTimeout(timeout)
  }, [scrollTrigger]);

  return useCallback(() => setScrollTrigger(Math.random()), []);
}
```

Then in the component:
```tsx
   const scrollToPackageList = usePackageListScrollTrigger(packageListRef.current)

   const createCollectionCardHandler = useCallback(
    (query: string) => () => {
      setIntegrationSearch(query);
      scrollToPackageList();
    },
    []
  );
```

Reply:
I'm not sure whether this is actually much nicer in terms of code - what is the `Math.random` to re-trigger an effect is adding? Seems like just calling it comes down to the same behavior and is more straightforward.

Reply:
It can be random() or just increment, main idea is to isolate this somewhere (custom hook seems like a good fit) and to have the timeout inside `useEffect` in order to be able to cancel the timeout when component unmounts. Maybe it's just a personal preference, I don't mind leaving it as is, but I think we still should handle the timeout cancelation properly.